### ISSUE TITLE
Include a section on gradient-free optimization in the README file

### ISSUE THREAD

**Issue by qthequartermasterman** [qthequartermasterman/torch_pso#20]
**Is your feature request related to a problem? Please describe.**
Currently, the README file is not clear about gradient-free optimization and why back-propagation isn't necessary in a gradient-free optimizer. This section should include commentary on using torch_pso with and without calling `loss.backward()` and its side-effects.

**Comment by qthequartermasterman** [comment]
Inspired by #17.
